rest/route/github.go

```diff
@@ -696,7 +696,28 @@ func (gh *githubHookApi) refreshPatchStatus(ctx context.Context, owner, repo str
 // both updates from patches and be in a race condition for which one GitHub checks shows last- so we want
 // to avoid this state when possible.
 func (gh *githubHookApi) AddIntentForPR(ctx context.Context, pr *github.PullRequest, owner, calledBy string, overrideExisting bool) error {
-	ghi, err := patch.NewGithubIntent(gh.msgID, owner, calledBy, pr)
+	// Verify that the owner/repo uses PR testing before inserting the intent.
+	baseRepoName := pr.Base.Repo.GetFullName()
+	baseRepo := strings.Split(baseRepoName, "/")
+	projectRef, err := model.FindOneProjectRefByRepoAndBranchWithPRTesting(baseRepo[0],
+		baseRepo[1], pr.Base.GetRef(), calledBy)
+	if err != nil {
+		return errors.Wrap(err, "finding project ref for patch")
+	}
+	if projectRef == nil {
+		return nil
+	}
+	var mergeBase string
+	if projectRef.OldestAllowedMergeBase != "" {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+		baseOwnerAndRepo := strings.Split(pr.Base.Repo.GetFullName(), "/")
+		mergeBase, err = thirdparty.GetGithubMergeBaseRevision(ctx, "", baseOwnerAndRepo[0], baseOwnerAndRepo[1], pr.Base.GetRef(), pr.Head.GetRef())
```

Comment:
We should check the length before accessing  baseOwnerAndRepo[1] to make sure it doesn't panic.